### DISCUSSION TITLE
Seria o Deno um Node.js "killer"?

### DISCUSSION THREAD

**Discussion by gustavofsantos** [frontendbr/forum#1294] [Perguntas e Respostas]
Olá, eu ando acompanhando o projeto que o Ryan Dahl vem trabalhando, chamado [deno](https://github.com/denoland/deno) desde quando ele o apresentou na [JSConf EU](https://www.youtube.com/watch?v=M3BM9TB-8yA) desse ano. Desde o início o projeto já evoluiu muito e já estou usando para escrever alguns utilitários que uso pra mim mesmo, pra automatizar o meu dia e também para testar o deno.

Alguém já conhecia?

E mais, vocês acham que realmente pode algum dia esse projeto chegar ao nível de uso que o Node tem hoje? Digo o ecossistema criado em cima do Node, que envolve todos os pacotes, libs, frameworks, etc.

Por fim, a pergunta mais crítica: vocês acham que um runtime que usa Typescript pode ganhar o :heart: de quem ainda não se acostumou com o TS, mesmo que seja só uma extensão para o JS?

**Comment by felipefialho** [comment]
Eu adoro concorrência de libs. Acho importante para o desenvolvimento da comunidade e correção de falhas.

Mas a questão do Node é profunda, a consolidação é real, e empresas que torciam o nariz para stack toda em JavaScript hoje aceitam com tranquilidade o uso. 

Ou seja, seria necessário uma adoção em larga escala, tanto pela comunidade quanto pelas empresas para que esse reinado fosse "destruido".

**Comment by ninetails** [comment]
Não conhecia mas não acho que substituiria. E acho mais fácil a galera migrar pra TS por causa de Angular ou da comunidade React usando TS crescer que ter mais uma engine. Ou a comunidade em volta do VS Code crescer ou todo mundo usar [ReactXP](https://microsoft.github.io/reactxp/)... sei lá

Superficialmente diria que se o deno conseguir um backer de peso, iria acontecer algo como já ocorreu:
- quando o Facebook lançou o HHVM pro PHP
- quando o Facebook lançou Yarn
- React vs Preact/Inferno/etc

Ambos coexistem e quando um lança uma funcionalidade melhor, o outro copia/adapta e melhora o próprio produto sem perder funcionalidades anteriores.

**Comment by gustavofsantos** [comment]
Concordo com vocês sobre a adoção de ferramentas e, um grande porém do deno é a incompatibilidade com todo o ecossistema que já existe em torno do Node. Mas espero que o projeto vá pra frente porque a gente não vai usar Node pra sempre

**Comment by wkrueger** [comment]
Deno parece mais um “side project conceitual” do Dahl, sei lá se realmente daria em alguma coisa, apesar do hype.
